src/commons/s3.rs

```diff
@@ -77,12 +77,15 @@ pub struct InlinedS3BucketSpec {
 
 impl InlinedS3BucketSpec {
     /// Build the endpoint URL from [S3ConnectionSpec::host] and [S3ConnectionSpec::port] and the S3 implementation to use
-    pub fn endpoint(&self, implementation: &S3ConnectionImplementation) -> Option<String> {
-        let implementation = implementation.to_string();
+    pub fn endpoint(&self) -> Option<String> {
```

Comment:
I want an endpoint function on the S3ConnectionSpec please

Reply:
No you do not want that (most likely) since endpoints only support http(s)

Reply:
Done.